packages/dd-trace/src/profiling/profilers/heap.js

```diff
@@ -1,20 +1,49 @@
 'use strict'
 
+const kStartSampleRate = 512 * 1024
+const kDefaultThreshold = 0.25
+
+function totalBytes (profile) {
+  let bytes = 0
+  for (const sample of profile.sample) {
+    bytes += sample.value[1]
+  }
+  return bytes
+}
+
 class NativeHeapProfiler {
   constructor (options = {}) {
     this.type = 'space'
-    this._samplingInterval = options.samplingInterval || 512 * 1024
+    this._samplingInterval = options.samplingInterval || kStartSampleRate
+    this._threshold = options.samplingThreshold || kDefaultThreshold
```

Comment:
Hey! I'm slightly confused here on the units -- I'm guessing the `_threshold` is a percentage between [0, 1]; but hat does `_samplingInterval` mean? 512*1024 seems like a weird value. Is it bytes? E.g. are we sampling every 512kb allocated?

It's a bit confusing to me since in `profile` we also talk about a rate of sampling in time :)

Reply:
V8 is only able to set the heap profiler sample rate by bytes. When n bytes have been allocated it will capture the current stack and how much it has allocated. Apps can be highly variable in how much they allocate though so rather than setting a constant sample rate in bytes which will produce a ton of data for some apps and very little for others, I'm essentially using this dynamic adjustment to try and roughly follow a time-based sample rate. It's not going to be 100% perfect as it could vary how much is allocated between requests, but it will try to stay roughly inline with a set time frequency for sampling. This should prevent excess data from slowing down the pprof encode too significantly.

Reply:
Thanks for the explanation! Would it make sense to add it as a comment to the class? It seems a quite valuable insight :)